### PR TITLE
Implement Command History

### LINKED ISSUE #137: Store past command in txt file

### LINKED ISSUE #138: Read up arrow or down arrow for next/previous commands

### LINKED ISSUE #144: Read from txt file past commands

### LINKED ISSUE #157: Clear text file upon exiting app
For command history

### LINKED ISSUE #163: Create test cases for command history

### PR DESCRIPTION
**Pull request by winstonlimjy**
- Write to text file upon every command. Fixes #137.
- Up-arrow key signifies previous command. Fixes #138.
- Down-arrow key signifies next command. 
- Read from text file upon up/down arrow key. Fixes #144.
- Clear text file upon exit. Fixes #157.
- Create relevant test cases. Fixes #163

### CODE REVIEW

File: src/main/java/seedu/address/logic/commands/NextCommandHistoryCommand.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+/**
+ * Navigate to the next command. Detected via down-arrow key.
+ */
+public class NextCommandHistoryCommand extends Command {
+    public static final String COMMAND_WORD = "downCommand";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Views the next command entered. ";
+
+    public static final String MESSAGE_SUCCESS = "Next command is shown";
+    @Override
```
Comment (gohsl99) [inline src/main/java/seedu/address/logic/commands/NextCommandHistoryCommand.java:17]: Code style: consider to remove line 13 and 15, and add new line between line 16 and 17

File: src/main/java/seedu/address/logic/commands/PreviousCommandHistoryCommand.java [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+import seedu.address.model.Model;
+
+/**
+ * Navigate to the previous command. Detected via up-arrow key.
+ */
+public class PreviousCommandHistoryCommand extends Command {
+    public static final String COMMAND_WORD = "upCommand";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Views the previous command entered. ";
+
```
Comment (gohsl99) [inline src/main/java/seedu/address/logic/commands/PreviousCommandHistoryCommand.java:15]: Code style: consider remove this line.

File: src/main/java/seedu/address/model/ModelManager.java [hunk 3]
```diff
@@ -176,6 +177,15 @@ public void setLastViewedClient(Client client) {
         this.lastViewedClient.set(client);
     }
 
+    //=========== Command History =================================================================
+    @Override
+    public String getPreviousCommand() {
+        return commandHistoryStorage.getPreviousCommand();
+    }
+    @Override
```
Comment (gohsl99) [inline src/main/java/seedu/address/model/ModelManager.java:185]: Code style: consider add new line between line 184 and 185.

File: src/main/java/seedu/address/model/ModelManager.java [hunk 4]
```diff
@@ -176,6 +177,15 @@ public void setLastViewedClient(Client client) {
...
+    public String getNextCommand() {
+        return commandHistoryStorage.getNextCommand();
+    }
     @Override
```
Comment (gohsl99) [inline src/main/java/seedu/address/model/ModelManager.java:189]: Code style: consider add new line between line 188 and 189.

File: src/main/java/seedu/address/storage/CommandHistoryStorage.java [hunk 5]
```diff
@@ -0,0 +1,183 @@
...
+
+/**
+ * Storage to store the past commands.
+ */
+public class CommandHistoryStorage {
+    private static Path commandHistoryFilePath = Paths.get("data", "commandHistory.txt");
+    private static ArrayList<String> lines = new ArrayList<>();
+    private int currentLineNumber;
+    private int previousTotalLines;
+    /**
```
Comment (gohsl99) [inline src/main/java/seedu/address/storage/CommandHistoryStorage.java:20]: Code style: consider adding new line between line 19 and 20.

File: src/test/java/seedu/address/logic/commands/NextCommandHistoryCommandTest.java [hunk 6]
```diff
@@ -0,0 +1,179 @@
...
+
+        public Client getLastViewedClient() {
+            throw new AssertionError("This method should not be called.");
+        }
+
+        @Override
+        public void setLastViewedClient(Client client) {
+            throw new AssertionError("This method should not be called.");
+        }
+        @Override
```
Comment (gohsl99) [inline src/test/java/seedu/address/logic/commands/NextCommandHistoryCommandTest.java:169]: Code style: consider adding new line between line 168 and 169.

File: src/test/java/seedu/address/logic/parser/AddressBookParserTest.java [hunk 7]
```diff
@@ -71,6 +73,19 @@ public void parseCommand_edit() throws Exception {
 
     //TODO: parseCommand_find()
 
+    @Test
+    public void parsePrevCommandHistory() throws ParseException {
+        PreviousCommandHistoryCommand previousCommandHistoryCommand =
+                (PreviousCommandHistoryCommand) parser.parseCommand("upCommand");
+        assertEquals(new PreviousCommandHistoryCommand(), previousCommandHistoryCommand);
+    }
+    @Test
```
Comment (gohsl99) [inline src/test/java/seedu/address/logic/parser/AddressBookParserTest.java:82]: Code style: consider adding new line between line 81 and 82.

**Review by gohsl99 — COMMENTED** [review]
LGTM, minor code style issue (adding new lines and redundant new lines).

File: src/main/java/seedu/address/logic/commands/NextCommandHistoryCommand.java (hunk 1 above)
Comment (winstonlimjy) [reply src/main/java/seedu/address/logic/commands/NextCommandHistoryCommand.java]: Sounds good to me thanks

File: src/main/java/seedu/address/logic/commands/PreviousCommandHistoryCommand.java (hunk 2 above)
Comment (winstonlimjy) [reply src/main/java/seedu/address/logic/commands/PreviousCommandHistoryCommand.java:15]: Same goes here. Will update accordingly.

File: src/main/java/seedu/address/model/ModelManager.java (hunk 3 above)
Comment (winstonlimjy) [reply src/main/java/seedu/address/model/ModelManager.java:185]: Updated

**Pull request merged**